app/controllers/admin/claims_controller.rb

```diff
@@ -22,14 +22,16 @@ def show
   end
 
   def search
-    return unless params[:reference].present?
+    return unless params[:query].present?
 
-    claim = Claim.find_by(reference: params[:reference].upcase)
+    @claims = Claim::Search.new(params[:query]).claims.includes(:eligibility)
 
-    if claim
-      redirect_to(admin_claim_url(claim))
+    if @claims.blank?
```

Comment:
Don't forget; `blank?` is an ActiveSupport extension primarily intended for dealing with things that might be strings. Either `none?` or `empty?` is more appropriate here:

```suggestion
    if @claims.empty?
```